File: backend/src/ips_app/domain/models/role.py
from datetime import datetime, timezone
from typing import Optional, Dict, Any, List
from pydantic import BaseModel, Field
from ips_app.domain.models.permission import Permission


class Role(BaseModel):
    id: Optional[Any] = None
    name: str
    description: str = ""
    preferences: Dict[str, Any] = Field(default_factory=dict)
    is_default: bool = False

    permissions: List[Permission] = Field(default_factory=list)

    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    created_by: Optional[Any] = None
    updated_at: Optional[datetime] = None
    updated_by: Optional[Any] = None

File: backend/internal/domain/models/role.go
package domainmodels

import (
	"encoding/json"
	"time"
)

type Role struct {
	Id          int64
	Name        string
	Description string
	Preferences json.RawMessage
	IsDefault   bool

	CreatedAt time.Time
	CreatedBy *int64
	UpdatedAt *time.Time
	UpdatedBy *int64
	Version   int64
}
